src/amplitude-client.js

```diff
@@ -1552,4 +1598,33 @@ if (BUILD_COMPAT_2_0) {
  */
 AmplitudeClient.prototype.__VERSION__ = version;
 
+/**
+ * Determines whether or not to push call to this._q or invoke it
+ * @private
+ */
+AmplitudeClient.prototype._shouldDeferCall = function _shouldDeferCall() {
+  return this._pendingReadStorage || this._initializationDeferred;
+};
+
+/**
+ * Defers Initialization by putting all functions into storage until users
+ * have accepted terms for tracking
+ * @private
+ */
+AmplitudeClient.prototype._deferInitialization = function _deferInitialization(apiKey, opt_userId, opt_config, opt_callback) {
+  this._initializationDeferred = true;
+  this._q.push(['init', apiKey, opt_userId, opt_config, opt_callback]);
+};
+
+/**
+ * Enable tracking via logging events and dropping a cookie
+ * Intended to be used with the deferInitialization configuration flag
+ * @public
+ */
+AmplitudeClient.prototype.enableTracking = function enableTracking() {
+  // This will call init (which drops the cookie) and will run any pending tasks
```

Comment:
nvm figured it out

Reply:
i was intending to drop the cookie so that when init gets called again (bc it's in `this._q`), it will short circuit the `this._deferInitialization()` and not reset `this._initialization` to `true` (this doesn't really work in this commit, but is fixed in the one i just pushed bc `this._initializationDeferred` is explicitly set to `false` in `enableTracking()`, but _both_ are needed for it to work)